utilities/pipelines/e2eValidation/resourceDeployment/New-TemplateDeployment.ps1

```diff
@@ -63,6 +63,152 @@ function Get-ErrorMessageForScope {
     }
 }
 
+<#
+.SYNOPSIS
+Monitors the state of a deployment through completion.
+
+.DESCRIPTION
+Monitors the state of a deployment through completion. This function works around transient issues with deployment monitoring using the native cmdlets.
+
+.PARAMETER DeploymentScope
+Mandatory. The scope to fetch the deployment from (e.g. resourcegroup, tenant,...)
+
+.PARAMETER DeploymentName
+Mandatory. The name of the deployment to search for (e.g. 'storageAccounts-20220105T0701282538Z')
+
+.PARAMETER ResourceGroupName
+Optional. The resource group to search the deployment in, if the scope is 'resourcegroup'
+
+.EXAMPLE
+Start-MonitorDeploymentForScope -DeploymentScope 'resourcegroup' -DeploymentName 'storageAccounts-20220105T0701282538Z' -ResourceGroupName 'validation-rg'
+
+Get the deployment for resource group 'validation-rg' that has the name 'storageAccounts-20220105T0701282538Z'
+
+.EXAMPLE
+Start-MonitorDeploymentForScope -DeploymentScope 'subscription' -DeploymentName 'resourcegroups-20220106T0401282538Z'
+
+Get the deployment for the current subscription that has the name 'storageAccounts-20220105T0701282538Z'
+#>
+
+function Start-MonitorDeploymentForScope {
```

Comment:
Debate if it should be kept. Would help for cases where the deployment experiences in internal error and just stops reporting. Would then pick the deployment log up again and continue tracking...